jujugui/static/gui/src/test/test_bakery.js

```diff
@@ -0,0 +1,159 @@
+/**
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+describe('Bakery', function() {
+  var bakery, macaroon, utils, Y;
+
+  before(function (done) {
+    var modules = ['juju-env-bakery', 'juju-tests-utils', 'macaroon' ];
+    Y = YUI(GlobalConfig).use(modules, function (Y) {
+      utils = Y['juju-tests'].utils;
+      macaroon = Y['macaroon'];
+      done();
+    });
+  });
+
+  beforeEach(function () {
+    bakery = new Y.juju.environments.web.Bakery();
```

Comment:
we had discussed passing in a stubbed version of the webhandler so that you would be able to test the most important part, the external interactions.